counsel.el

```diff
@@ -5096,6 +5100,44 @@ build environments.")
                      (or (buffer-file-name) default-directory)
                      ".git")))
 
+;; This is loosely based on the bash completion code
+(defun counsel--get-make-calls (root)
+  "Return a list of make calls for a given directory.
+
+We search the Makefile for a list of PHONY targets which are generally
+the top-level targets a make system provides."
+  (let ((default-directory root))
+    (mapcar
+     (lambda(target)
+       (format "cd %s; make %s %s" root counsel-compile-make-opts target))
+     (split-string
+      (shell-command-to-string
+       (concat "make -nqp |"
+               "grep -B 1 PHONY |"
+               "grep ':' |"
+               "cut -d ':' -f 1 |"
+               "sort"))
```

Comment:
1.  What if `make` or any further commands return non-`0` exit status?  We don't want to deal with their `stderr` outputs, i.e. those have to be suppressed by redirection on or some other portable solution.
2.  Sorting via external process may be an overkill here.  Do we need it at all?